OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -766,7 +766,15 @@ static bool Kick(S server, Connection conn, Session.Client client, string s)
 				}
 
 				var kickClient = server.GetClient(kickConn);
-				if (server.State == ServerState.GameStarted && !kickClient.IsObserver)
+
+				var ws = kickClient.IsObserver
+					? WinState.Lost
+					: server.WorldPlayers
+						.Where(p => p != null && p.ClientIndex == kickClient.Index)
+						.Select(p => p.Outcome)
+						.Single();
```

Comment:
We can combine this with the suggestions from Discord:
```suggestion
				var winState = server.WorldPlayers.SingleOrDefault(p => p?.ClientIndex == kickClient.Index)?.Outcome;
```
and then check if `winState` is `null` or not `WinState.Lost`.